apis/v1beta1/config.go

```diff
@@ -90,6 +114,62 @@ type Config struct {
 	Service    Service    `json:"service" yaml:"service"`
 }
 
+// GetEnabledComponents constructs a list of enabled components by component type.
+func (c *Config) GetEnabledComponents() map[ComponentType]map[string]interface{} {
```

Comment:
This is a pretty simple change to break out, it will trim a few hundred lines too!